core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -1235,7 +1234,7 @@ class UnifiedLog(@volatile var logStartOffset: Long,
    *           <li>All special timestamp offset results are returned immediately irrespective of the remote storage.
    *         </ul>
    */
-  def fetchOffsetByTimestamp(targetTimestamp: Long, remoteLogManager: Option[RemoteLogManager] = None): OffsetResultHolder = {
+  def fetchOffsetByTimestamp(targetTimestamp: Long, remoteLogManager: Optional[AsyncOffsetReader] = Optional.empty): OffsetResultHolder = {
```

Comment:
nit: 

Can we rename the variable `remoteLogManager` to `remoteOffsetReader`? As, the other methods in RemoteLogManager are no longer accessible.

Reply:
Done